cross-chain/solana/solana-tbtc-anchor/tests/solana-tbtc-anchor.ts

```diff
@@ -0,0 +1,444 @@
+import * as anchor from "@coral-xyz/anchor";
+import { Program, AnchorError } from "@coral-xyz/anchor";
+import * as spl from "@solana/spl-token";
+import * as web3 from '@solana/web3.js';
+import { SolanaTbtcAnchor } from "../target/types/solana_tbtc_anchor";
+import { expect } from 'chai';
+
+async function setup(
+  program: Program<SolanaTbtcAnchor>,
+  tbtc,
+  creator
+) {
+  const [tbtcMintPDA, _] = getTokenPDA(program, tbtc);
+  
+  await program.methods
+    .initialize()
+    .accounts({
+      tbtcMint: tbtcMintPDA,
+      tbtc: tbtc.publicKey,
+      creator: creator.publicKey
+    })
+    .signers([tbtc])
+    .rpc();
+}
+
+async function checkState(
+  program: Program<SolanaTbtcAnchor>,
+  tbtc,
+  expectedCreator,
+  expectedMinters,
+  expectedGuardians,
+  expectedTokens
+) {
+  let tbtcState = await program.account.tbtc.fetch(tbtc.publicKey);
+
+  expect(tbtcState.creator).to.eql(expectedCreator.publicKey);
+  expect(tbtcState.minters).to.equal(expectedMinters);
+  expect(tbtcState.guardians).to.equal(expectedGuardians);
+
+  let tbtcMint = tbtcState.tokenMint;
+
+  let mintState = await spl.getMint(program.provider.connection, tbtcMint);
+
+  expect(mintState.supply == expectedTokens).to.be.true;
+}
+
+async function checkPaused(
+  program: Program<SolanaTbtcAnchor>,
+  tbtc,
+  paused: boolean
+) {
+  let tbtcState = await program.account.tbtc.fetch(tbtc.publicKey);
+  expect(tbtcState.paused).to.equal(paused);
+}
+
+function getTokenPDA(
+  program: Program<SolanaTbtcAnchor>,
+  tbtc,
+): [anchor.web3.PublicKey, number] {
+  return web3.PublicKey.findProgramAddressSync(
+    [
+      anchor.utils.bytes.utf8.encode('tbtc-mint'),
+      tbtc.publicKey.toBuffer(),
+    ],
+    program.programId
+  );
+}
+
+function getMinterPDA(
+  program: Program<SolanaTbtcAnchor>,
+  tbtc,
+  minter
+): [anchor.web3.PublicKey, number] {
+  return web3.PublicKey.findProgramAddressSync(
+    [
+      anchor.utils.bytes.utf8.encode('minter-info'),
+      tbtc.publicKey.toBuffer(),
+      minter.publicKey.toBuffer(),
+    ],
+    program.programId
```

Comment:
The Minter PDA is the address of the `MinterInfo` account, derived from the TBTC program ID. If you want to use a WormholeGateway PDA for the minting authority, you would supply that PDA as the `minter` so it can sign transactions as the authority of the corresponding `MinterInfo`.

Reply:
It is because the minter PDA is for the minter info account. The minter info account contains the pubkey of the `minter` address that actually holds minting authority. The `minter` address is the one that is a signer for minting instructions. For `WormholeGateway` to have that authority it is this `minter` address that needs to be a PDA derived from `WormholeGateway`. The `MinterInfo` account never signs anything, it only contains data.

The `minterKeys` in tests is the address that would be replaced with a PDA derived from `WormholeGateway` as it is that address which signs the minting transactions. As part of the minting transaction we check the `MinterInfo` account, which does not sign anything, only contains data for tbtc, that the `minter` address contained in the `MinterInfo` account is the signer of the minting transaction.

Reply:
Ok, that makes sense, thx.